src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,794 @@
+        #region Options
+        internal sealed class Options : TransformInputBase
+        {
+            [Argument(ArgumentType.Required, HelpText = "Input column", Name = "Source", ShortName = "src", SortOrder = 1)]
+            public string Source;
+
+            // This transformer adds columns
+            [Argument(ArgumentType.Required, HelpText = "Output column prefix", Name = "Prefix", ShortName = "pre", SortOrder = 2)]
+            public string Prefix;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Country to get holidays for. Defaults to none if not passed", Name = "Country", ShortName = "ctry", SortOrder = 4)]
+            public HolidayList Country = HolidayList.None;
+        }
+
+        #endregion
+
+        internal DateTimeEstimator(IHostEnvironment env, string inputColumnName, string columnPrefix, HolidayList country = HolidayList.None)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+            _host.CheckValue(inputColumnName, nameof(inputColumnName), "Input column should not be null.");
+
+            _options = new Options
+            {
+                Source = inputColumnName,
+                Prefix = columnPrefix,
+                Country = country
+            };
+        }
+
+        internal DateTimeEstimator(IHostEnvironment env, Options options)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+
+            _options = options;
+        }
+
+        public DateTimeTransformer Fit(IDataView input)
+        {
+            return new DateTimeTransformer(_host, _options.Source, _options.Prefix, _options.Country);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            var columns = inputSchema.ToDictionary(x => x.Name);
+
+            foreach (ColumnsProduced column in Enum.GetValues(typeof(ColumnsProduced)))
+            {
+                columns[_options.Prefix + column.ToString()] = new SchemaShape.Column(_options.Prefix + column.ToString(), SchemaShape.Column.VectorKind.Scalar,
+                ColumnTypeExtensions.PrimitiveTypeFromType(column.GetRawColumnType()), false, null);
+            }
+
+            return new SchemaShape(columns.Values);
+        }
+
+        #region Enums
+        public enum ColumnsProduced : byte
+        {
+            Year = 1,
+            Month = 2,
+            Day = 3,
+            Hour = 4,
+            Minute = 5,
+            Second = 6,
+            AmPm = 7,
+            Hour12 = 8,
+            DayOfWeek = 9,
+            DayOfQuarter = 10,
+            DayOfYear = 11,
+            WeekOfMonth = 12,
+            QuarterOfYear = 13,
+            HalfOfYear = 14,
+            WeekIso = 15,
+            YearIso = 16,
+            MonthLabel = 17,
+            AmPmLabel = 18,
```

Comment:
If we do AmPm label, would it make sense to also do weekend/weekday? (Often used in feature engineering)

Reply:
I think thats not a bad idea, but the initial ask we got was just for these columns. We will start with just these and then can add in more if the requirement arises.